diff -aurN libevent-2.0.20-stable.orig/Makefile.am libevent-2.0.20-stable/Makefile.am
--- libevent-2.0.20-stable.orig/Makefile.am	2012-09-17 21:57:27.424825262 +0200
+++ libevent-2.0.20-stable/Makefile.am	2012-09-17 21:58:52.932184801 +0200
@@ -197,17 +197,17 @@
 
 if PTHREADS
 libevent_pthreads_la_SOURCES = evthread_pthread.c
-libevent_pthreads_la_LIBADD = $(MAYBE_CORE)
+libevent_pthreads_la_LIBADD = $(MAYBE_CORE) libevent.la
 libevent_pthreads_la_LDFLAGS = $(GENERIC_LDFLAGS)
 endif
 
 libevent_extra_la_SOURCES = $(EXTRA_SRC)
-libevent_extra_la_LIBADD = $(MAYBE_CORE) $(SYS_LIBS)
+libevent_extra_la_LIBADD = $(MAYBE_CORE) $(SYS_LIBS) libevent.la
 libevent_extra_la_LDFLAGS = $(GENERIC_LDFLAGS)
 
 if OPENSSL
 libevent_openssl_la_SOURCES = bufferevent_openssl.c
-libevent_openssl_la_LIBADD = $(MAYBE_CORE) $(OPENSSL_LIBS)
+libevent_openssl_la_LIBADD = $(MAYBE_CORE) libevent.la $(OPENSSL_LIBS)
 libevent_openssl_la_LDFLAGS = $(GENERIC_LDFLAGS)
 endif
 
